File: pset6/credit/credit.py
from cs50 import get_int
from sys import exit

h = get_int("Enter the number:   ")
s = str(h)
l = len(s)
sum = 0
count = 0

for i in s[::-1]:
    i = int(i)
    if count % 2 != 0:
        i *= 2
        if i > 9:
            i -= 9
            sum += i
        else :
            sum += i
        count += 1
    else :
        sum += i
        count += 1

if sum % 10 != 0:
    print("Invalid card")
    exit(1)

check = s[0:2]

if (check in ["34", "37"]) and (l == 15):
    print("AMEX")

elif (check in ["54", "51", "52", "53", "55"]) and (l == 16):
    print("MASTERCARD")

elif (s[0] == "4") and (l in [13, 16]):
    print("VISA")

else:
    print("unknown")

File: pset1/credit.c
#include <stdio.h>
#include <cs50.h>

int count (int n);

int main (void)
{
    long int h;
    do
    {
        h = get_long("Enter the Number:  ");
    }
    while (h <= 0);

    // making a copy of the card no.
    long n = h;

    // counting no. of digits

    int count = 0, l;
    while (h != 0)
    {
        h /= 10;
        if (count == 14)
        {
            l = h;
        }
        count++;
    }
    int sum =0, digit, digitnum=1;
    while (n != 0)
    {
        digit = n % 10;
        n /= 10;
        if (digitnum % 2 == 0)
        {
            digit *= 2;
            if (digit > 9)
            {
                digit -= 9;
            }
            sum += digit;
        }
        else
        {
            sum += digit;
        }
        digitnum ++;

    }
    if (sum % 10 != 0)
    {
        printf("INVALID\n");
    }
    if(count == 15)
    {
        printf("AMEX\n");
    }
    else if(count == 13)
    {
        printf("VISA\n");
    }
    else if (count == 16 && l == 4)
    {
        printf("VISA\n");
    }
    else if (count == 16)
    {
        printf("MASTERCARD\n");
    }

}
